pkg/server/endpoints/endpoints.go

```diff
@@ -89,6 +89,10 @@ type RateLimitConfig struct {
 
 // New creates new endpoints struct
 func New(ctx context.Context, c Config) (*Endpoints, error) {
+	if err := os.MkdirAll(c.UDSAddr.String(), 0750); err != nil {
```

Comment:
awesome sauce. Does the agent already do this too?

Reply:
It does, although in a different spot (in the agent run cmd).